test_cottage_page.js

```diff
@@ -0,0 +1,155 @@
+								return parseInt(document.getElementsByClassName("fulldescription")[0].style.height.split("px")[0], 10);
+					});
+					test.assertTrue(tmp_height < new_height  , "Check that the read more button 'un-expands' the description");
+				});
+			});
+		}).then(function() {
+			//Verify that content tabs work;
+			//Reviews content tab;
+			test.comment("Click reviews tab.");
+			casper.click("a[href='#reviews']");
+			casper.then(function() {
+					var state = this.evaluate(function() {
+						return document.getElementById("reviews").className;
+					});
+					test.assertTrue(state.indexOf("showing") > -1, "Check that the reviews panel has class 'showing'.");
+					test.assertVisible("#reviews", "Check that the reviews panel is visible.");
+					test.comment("Click Prices and availability tab");
+					casper.click("a[href='#prices-and-availability']");
+				}).then(function() {
+					var state = this.evaluate(function() {
+						return document.getElementById("prices-and-availability").className;
+					});
+					test.assertTrue(state.indexOf("showing") > -1, "Check that the prices and availability panel has class 'showing'.");
+					test.assertVisible("#prices-and-availability", "Check that the prices-and-availability panel is visible.");
+					test.comment("Click town-village-guide tab");
+					casper.click("a[href='#town-village-guide']");
+				}).then(function() {
+					var state = this.evaluate(function() {
+						return document.getElementById("town-village-guide").className;
+					});
+					test.assertTrue(state.indexOf("showing") > -1, "Check that the town and village guide tab has class 'showing'.");
+					test.assertVisible("#town-village-guide", "Check that the town-village-guide is visible.");
+					test.comment("Click property-overview tab");
+					casper.click("a[href='#property-overview']");
+				}).then(function() {
+					var state = this.evaluate(function() {
+						return document.getElementById("property-overview").className;
+					});
+					test.assertTrue(state.indexOf("showing") > -1, "Check that the property-overview tab has class 'showing'.");
+					test.assertVisible("#property-overview", "Check that the property-overview panel is visible.");
+				});
+		}).then(function() {
+			//Check shortlist functionality.
+			test.comment("Clicking add to shortlist button.");
+			casper.click("a.add-to-shortlist");
+			casper.waitForSelectorTextChange("a.add-to-shortlist", function(e) {
+				var btn = casper.evaluate(function() {
+					return document.getElementsByClassName("add-to-shortlist")[0].innerText.replace(" ", "");
+				});
+				test.assertTrue(btn == "Remove", "Check shortlist button text has changed to 'remove'.")
+			});
+		}).then(function() {
+			test.comment("Ascertaining whether property is on shortlist page.");
+			casper.clickLabel("Your shortlist");
+			casper.then(function() {
+				test.assertTrue(this.getCurrentUrl() == url + "shortlist/", "Check that shortlist page has loaded correctly.");
```

Comment:
Try to avoid unnecessary nesting, this way would reduce nesting:

then(function(){
casper.clickLabel("Your shortlist");
}).then(function(){
  test.assertTrue(this.getCurrentUrl() == url + "shortlist/", "Check that shortlist page has loaded correctly.");
  ...
})